app/models/post.rb

```diff
@@ -1,5 +1,25 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :increment
+  after_destroy :decrement
+
+  attribute :comments_counter, :integer, default: 0
```

Comment:
* * Kindly fix this error i get when i try to create comments using the `rails console`  `arguments(given: 2, expect: 0).`  HINT 💭 you made a mistake setting the `:comments_counter` instead of `:posts_counter` also kindly add the attribute for in your `comment.rb model file` below is a screenshot of the correction i made and the comment_counter working 
![image](https://user-images.githubusercontent.com/52497246/183568390-9a261fe2-662c-499f-b333-c622d998b364.png)

Reply:
@Davidosky007 Hello, David! Thanks a lot for showing, but I tried doing that and I'm a bit confused by the instructions, would mind showing what you changed in the comments, user and like models as well?

Reply:
@Davidosky007 Never mind, I fixed it, but it wasn't the attribute, it was the name of the methods increment and decrement being the same all over the models that were giving the problem